spikeinterface/comparison/basecomparison.py

```diff
@@ -88,19 +88,19 @@ def _do_comparison(self, ):
         if self._verbose:
             print('Multicomaprison step 1: pairwise comparison')
 
-        self.comparisons = []
+        self.comparisons = {}
         for i in range(len(self.object_list)):
             for j in range(i + 1, len(self.object_list)):
+                if self.name_list is not None:
+                    name_i = self.name_list[i]
+                    name_j = self.name_list[j]
+                else:
+                    name_i = "object i"
+                    name_j = "object j"
                 if self._verbose:
-                    if self.name_list is not None:
-                        name_i = self.name_list[i]
-                        name_j = self.name_list[j]
-                    else:
-                        name_i = "object i"
-                        name_j = "object j"
                     print(f"  Comparing: {name_i} and {name_j}")
                 comp = self._compare_ij(i, j)
-                self.comparisons.append(comp)
+                self.comparisons[(name_i, name_j)] = comp
```

Comment:
Sure there is no side effect somewhere else switching list to dict ?

Reply:
No I checked. It's only used internally ;)